shuup/admin/utils/product.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.db.transaction import atomic
+
+from shuup.utils.models import copy_model_instance
+from shuup.core.models import (
+    Product, ProductAttribute, ProductMedia, ShopProduct
+)
+from shuup.admin.signals import product_copied
+from shuup.utils.models import get_data_dict
+
+
+class ProductCloner():
+    def __init__(self, current_shop, current_supplier):
+        self.current_shop = current_shop
+        self.current_supplier = current_supplier
+
+    @atomic()
+    def clone_product(self, shop_product=None):
+        # Clone product
+        product = shop_product.product
+        new_product = copy_model_instance(product)
+        new_product.sku = "{}-{}".format(product.sku, Product.objects.count())
+        new_product.name = ("{name} - Copy").format(name=product.name)
+        new_product.save()
+
+        # Change the name so it will automatically create a translation record
+        for trans in product.translations.all()[1:]:
```

Comment:
why are you using this `[1:]`? Can't you just use `get_or_create` below instead?

Reply:
As I check, when we modified new_product's name in line 29, it will create a new translation records and add it to new_product. So I skip the first one